stable/spark-history-server/templates/deployment.yaml

```diff
@@ -60,31 +69,48 @@ spec:
           httpGet:
             path: /
             port: historyport
-        volumeMounts:
         {{- if .Values.pvc.enablePVC }}
+        volumeMounts:
         - name: data
           mountPath: /mnt/{{ .Values.pvc.eventsDir }}
         {{- else if .Values.gcs.enableGCS }}
+        volumeMounts:
         - name: secrets-volume
           mountPath: /etc/secrets
+        {{- else if .Values.s3.enableS3 }}
+        {{- if (not .Values.s3.enableIAM) }}
```

Comment:
Can be simplified to {{- else if and (.Values.s3.enableS3) (not .Values.s3.enableIAM) }}

Reply:
This was done intentionally. As mentioned above, there are two cases within when s3 is enabled: (1) IAM authentication is used (2) IAM authentication is NOT used and fall back to using access key id and secret key. I only need the secret volume to be mounted in the later case and the former case does not need any of that.
If I did `{{- else if and (.Values.s3.enableS3) (not .Values.s3.enableIAM) }}`, the former case will go to `else` block to mount `hdfs-site` and `core-site`.

There was no flag to indicate if you are using `hdfs-site` and `core-site`, and I didn't want to introduce a new flag there without talking to you.